1-source-code/3-reading/2-guru.md

```diff
@@ -0,0 +1,51 @@
+# Tools for navigating code
+
+Source code in any language is a very rich structure with many links
+forward and backwards, every variable usage, every function call,
+refers to some object defined somewhere in the code base.
+
+Navigating this with the help of just `grep` or any other text search
+is far from optimal, but some tools exist specifically for this.
+
+## godef
+
+`godef` allows you to navigate those links in a specific direction,
```

Comment:
it's installed with VSCode, and calling it manually is not *that* simple or useful imo